### PR TITLE
Remove usage of concurrent collections from rules

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**
As for the other performance tests I did 4 runs on Akka, EfCore and Noda projects.  The sum of the execution time of the 3 rules changed, RequestsWithExcessiveLength, SetLocaleForDataTypes and UnusedPrivateMember with and without the changes are the following:
akka:  28.47 -> 32.893
efcore: 36.698 -> 37.362
noda: 15.455 -> 13.409

### CODE REVIEW

**Comment by Corniel** [comment]
I assume this is done to speed up things. Do you already have measurements on it? (I'm just curious)

**Comment by csaba-sagi-sonarsource** [comment]
We don't yet have any numbers, the measurements will be done at some point this week.

**Comment by Corniel** [comment]
> We don't yet have any numbers, the measurements will be done at some point this week.
I hope you can share them.

File: analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs [hunk 1]
```diff
@@ -118,14 +117,13 @@ private void CollectAttributesOverTheLimit(SyntaxNodeAnalysisContext context, Co
...
-                    new Attributes(requestFormLimits, requestSizeLimit),
-                    (_, attributes) => new Attributes(requestFormLimits, requestSizeLimit, attributes));
+                attributesOverTheLimit[declaration] = attributesOverTheLimit.TryGetValue(declaration, out var existingAttribute)
+                    ? new Attributes(requestFormLimits, requestSizeLimit, existingAttribute)
+                    : new Attributes(requestFormLimits, requestSizeLimit);
             }
         }
 
-        private void ReportOnCollectedAttributes(CompilationAnalysisContext context, ConcurrentDictionary<SyntaxNode, Attributes> attributesOverTheLimit)
+        private void ReportOnCollectedAttributes(CompilationAnalysisContext context, Dictionary<SyntaxNode, Attributes> attributesOverTheLimit)
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs:126]: Can be `IDictionary<SyntaxNode, Attributes>`

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**